### PR TITLE
CSV filtering module Coverage 30%

### PR DESCRIPTION
**Pull request by Juerodriguez**
## Summary

Create this script to filter the data from a csv by loading the instance, charter, jurisdiction tables that contain a column.

## Checklist

- [x] Do script
- [x] Python code with PEP8

## Screenshots

(Quick views, image png, gif  and/or video from Playwright)

### CODE REVIEW

File: modules/filtro_carga_csv/filtro.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import csv
+import collections
+import argparse
+
+data = []
```
Comment (eamanu) [inline modules/filtro_carga_csv/filtro.py:5]: se suele usar mayusculas para virables globales

File: modules/filtro_carga_csv/filtro.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+data_in_column = []
+
+def filtered_csv(file_csv, ent_column):
+    global data
```
Comment (eamanu) [inline modules/filtro_carga_csv/filtro.py:9]: sin `global` no funciona?

File: modules/filtro_carga_csv/filtro.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    global data_in_column
+    with open(file_csv, newline = '') as File:
```
Comment (eamanu) [inline modules/filtro_carga_csv/filtro.py:11]: `File` debería estar en minuscula para seguir pep8

Comment (eamanu) [reply modules/filtro_carga_csv/filtro.py:11]: igualmente file es una palabra reservada.. deberías pensar en otra variable

File: modules/filtro_carga_csv/filtro.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        input_data = csv.DictReader(File)
+        for file in input_data:
+            data.append(file[ent_column])
```
Comment (eamanu) [inline modules/filtro_carga_csv/filtro.py:14]: podrías usar list comprehension https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions
`data = [file[eng_column] for file in input_data]`

File: modules/filtro_carga_csv/filtro.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+def upload(out_csv):
+    with open(out_csv, "w", newline='') as file:
+        cargar = csv.writer(file)
+        cargar.writerows(data_in_column)
+    file.close()
+
+def parse_args():
+    parser = argparse.ArgumentParser(description = "Este Script sirve para filtrar los csv que contengan una lista de jurisdicciones")
+    parser.add_argument("file_csv", help = "Archivo de entrada csv")
```
Comment (eamanu) [inline modules/filtro_carga_csv/filtro.py:26]:
```suggestion
    parser.add_argument("file_csv", help="Archivo de entrada csv")
```

File: modules/filtro_carga_csv/filtro.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    parser.add_argument("ent_column", help = "Nombre de columna de jurisdicciones")
+    parser.add_argument("out_csv", help = "Archivo csv de salida")
```
Comment (eamanu) [inline modules/filtro_carga_csv/filtro.py:28]:
```suggestion
    parser.add_argument("ent_column", help="Nombre de columna de jurisdicciones")
    parser.add_argument("out_csv", help="Archivo csv de salida")
```

**Review by eamanu — CHANGES_REQUESTED** [review]

File: modules/filtro_carga_csv/filtro.py (hunk 2 above)
Comment (Juerodriguez) [reply modules/filtro_carga_csv/filtro.py:9]: Tengo que modificar eso ya que la unica variable global era data_in_column.
La cual no funciona sin llamarla en global

Comment (eamanu) [reply modules/filtro_carga_csv/filtro.py:9]: Entonces dejalo como está.

**Comment by eamanu** [comment]
Me parece más indicado colocar este script en alguna carpeta llamada "scripts", no?

cc:  @marian-vignau

File: modules/filtro_carga_csv/filtro.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import csv
+import collections
+import argparse
+
+DATA_IN_COLUMN = []
+
+
+def filtered_csv(file_csv, ent_column):
+    global data_in_column
```
Comment (eamanu) [inline modules/filtro_carga_csv/filtro.py:9]: seems that this isn't longer need?

**Review by eamanu — CHANGES_REQUESTED** [review]
Please add some unittests

File: modules/filtro_carga_csv/filtro.py (hunk 7 above)
Comment (eamanu) [reply modules/filtro_carga_csv/filtro.py:9]: ping @Juerodriguez

Comment (Juerodriguez) [reply modules/filtro_carga_csv/filtro.py:9]:
```suggestion
    global DATA_IN_COLUMN
```
Deberia quedar asi

File: modules/filtro_carga_csv/filtro.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        data = [file[ent_column] for file in input_data]
+        key_data = list(collections.OrderedDict.fromkeys(data))
+        DATA_IN_COLUMN = [[row] for row in key_data]
+
+
+def upload(out_csv):
+    with open(out_csv, "w", newline='') as file_:
+        uploading = csv.writer(file_)
+        uploading.writerows(DATA_IN_COLUMN)
+    file_.close()
```
Comment (eamanu) [inline modules/filtro_carga_csv/filtro.py:21]: file is already closed when `with` clause end. So this `file_.close()` isn't necessary. 
From documentation:
```
It is good practice to use the with keyword when dealing with file objects. The advantage is that the file is properly closed after its suite finishes, even if an exception is raised at some point. Using with is also much shorter than writing equivalent try-finally blocks...
```
https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files

**Review by eamanu — APPROVED** [review]

File: modules/filtro_carga_csv/filtro.py (hunk 8 above)
Comment (Juerodriguez) [reply modules/filtro_carga_csv/filtro.py:21]: Indeed

Comment (Juerodriguez) [inline modules/filtro_carga_csv/filtro.py:21]:
```suggestion

```

**Pull request merged**